src/ai/configuration.cpp

```diff
@@ -217,7 +217,7 @@ bool configuration::parse_side_config(side_number side, const config& original_c
 	DBG_AI_CONFIGURATION << "side " << side << ": config contains:"<< std::endl << cfg;
 
 	//insert default config at the beginning
-	if (default_config_) {
+	if (true) {
```

Comment:
`default_config_` (defined in the header file) is not a reference.

Reply:
No, I mean… if it's always true, why is there an if at all?

Reply:
probably because the person who write that code assumed that ` default_config_ = ais.child("default_config");` a few lines above would rebind the  default_config_ object instead of invoking `config::operator=`. One could fix that code, but i explicitly decided not to change behviour in this commit too much since its already complicated enough.

Reply:
Continued at https://github.com/wesnoth/wesnoth/pull/7497#discussion_r1155214338